platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/SimpleListItem.kt

```diff
@@ -12,23 +12,218 @@ import androidx.compose.runtime.Composable
+ * @param textModifier The [Modifier] applied to the list item text
+ * @param iconModifier Optional [Modifier] to be applied specifically to the icon.
+ * @param icon Optional [IconKey] representing the icon displayed in the list item.
+ * @param iconContentDescription Content description for accessibility purposes for the icon.
+ * @param style Optional [SimpleListItemStyle] for defining the appearance of the list item.
+ * @param height The height of the list item; default is based on the theme's global metrics.
+ * @param painterHints Optional vararg of [PainterHint] to provide hints for painting customizations.
+ */
+@Composable
+public fun SimpleListItem(
+    text: String,
+    state: ListItemState,
+    colorFilter: ColorFilter?,
+    modifier: Modifier = Modifier,
+    textModifier: Modifier = Modifier,
+    iconModifier: Modifier = Modifier,
+    icon: IconKey? = null,
+    iconContentDescription: String? = null,
+    style: SimpleListItemStyle = JewelTheme.simpleListItemStyle,
+    height: Dp = JewelTheme.globalMetrics.rowHeight,
+    vararg painterHints: PainterHint,
+) {
+    SimpleListItem(
+        state = state,
+        modifier = modifier,
+        iconModifier = iconModifier,
+        icon = icon,
+        iconContentDescription = iconContentDescription,
+        style = style,
+        height = height,
+        content = {
+            Text(
+                text = text,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                style = JewelTheme.defaultTextStyle,
+                color = style.colors.contentFor(state).value,
+                modifier = textModifier,
+            )
+        },
+        painterHints = painterHints,
+        colorFilter = colorFilter,
+    )
+}
+
+@Composable
+public fun SimpleListItem(
+    text: String,
+    selected: Boolean,
+    colorFilter: ColorFilter?,
+    modifier: Modifier = Modifier,
+    textModifier: Modifier = Modifier,
+    iconModifier: Modifier = Modifier,
+    active: Boolean = true,
+    icon: IconKey? = null,
+    iconContentDescription: String? = null,
+    style: SimpleListItemStyle = JewelTheme.simpleListItemStyle,
+    height: Dp = JewelTheme.globalMetrics.rowHeight,
+    vararg painterHints: PainterHint,
+) {
+    val state = remember(selected, active) { ListItemState(selected, active) }
+    SimpleListItem(
+        text = text,
+        state = state,
+        modifier = modifier,
+        textModifier = textModifier,
+        iconModifier = iconModifier,
+        icon = icon,
+        iconContentDescription = iconContentDescription,
+        style = style,
+        height = height,
+        colorFilter = colorFilter,
+        painterHints = painterHints,
+    )
+}
+
+/**
+ * A simple list item layout comprising of a content slot and an optional icon to its start side.
+ *
+ * The text will only take up one line and is ellipsized if too long to fit. The item will draw a background based on
```

Comment:
This is the slot-based version, so this behaviour is not up to us at all. Please double check the KDocs, feels like it needs some cleanup after copy-pasting